projects/plugins/boost/app/modules/optimizations/critical-css/Generator.php

```diff
@@ -23,32 +23,17 @@ public function __construct() {
 	 * phpcs:disable WordPress.Security.NonceVerification.Recommended
 	 */
 	public static function is_generating_critical_css() {
-		static $is_generating = null;
-		if ( null !== $is_generating ) {
-			return $is_generating;
-		}
-
-		// Accept nonce via HTTP headers or GET parameters.
-		$generate_nonce = null;
-		if ( ! empty( $_GET[ self::GENERATE_QUERY_ACTION ] ) ) {
-			$generate_nonce = sanitize_key(
-				$_GET[ self::GENERATE_QUERY_ACTION ]
-			);
-		} elseif ( ! empty( $_SERVER['HTTP_X_GENERATE_CRITICAL_CSS'] ) ) {
-			$generate_nonce = sanitize_key(
-				$_SERVER['HTTP_X_GENERATE_CRITICAL_CSS']
-			);
-		}
-
 		// If GET parameter or header set, we are trying to generate.
-		$is_generating = ! empty( $generate_nonce );
+		if ( isset( $_GET[ self::GENERATE_QUERY_ACTION ] ) ) {
+			return true;
+		}
 
-		// Die if the nonce is invalid.
-		if ( $is_generating && ! Nonce::verify( $generate_nonce, self::GENERATE_QUERY_ACTION ) ) {
-			die();
+		// If there is an HTTP header indicating that we are generating, i.e. boost-cloud
+		if ( isset( $_SERVER['HTTP_X_GENERATE_CRITICAL_CSS'] ) ) {
+			return true;
```

Comment:
Note caching: if the page is cached and there is no GET param, for the boost cloud request, this is going to return the cached version of the page, which might have this variable set to `false`, even if the header is set. I haven't walked the call tree to understand what the effects are of `is_generating_critical_css`, but providing a boolean based on header for a generic public function like this can cause problems down the line, even if there are none now.

Reply:
I assumed this was for hydra. But, now that I double-checked it, apparently, hydra doesn't actually send this header. So, I am guessing we can remove this check and add the same query param-based check-in hydra. 

If `is_generating_critical_css()` is true, a meta tag is added to the page. The generator can look for this to ensure that it has reached the correct page and not some WAF or 404. Currently there is nothing like this for hydra and it would be quite useful.

Did I miss any usage of this header? You introduced it a while ago.